src/core/provider/CustomJsonRpcProvider.ts

```diff
@@ -29,4 +43,8 @@ export class CustomJsonRpcProvider extends JsonRpcProvider {
       currentBlockNumber - (AVERAGE_BLOCKS_PER_10_MINUTES[this.chainId] ?? 0) // default to 0 to avoid failures
     return blockNumberTenMinsAgo
   }
+
+  async call(transaction: TransactionRequest): Promise<string> {
+    return this.retryHandler(() => super.call(transaction))
+  }
 }
```

Comment:
can you add the retry handler on getLogs aswel please?

Reply:
i wouldn't wait 20 second for logs though the request should be resolved in a few seconds

Reply:
Added another `retryHandler` for `getLogs`.